Automattic/wp-calypso
# A4A: Making sure unreachable sites are clearly indicated and links blocked

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/225

## Proposed Changes

* The intention of this PR is to make it clearer in the sites dashboard when a site is unreachable.
* The definition of broken is a site (should at the moment be) whose Jetpack connection is not working properly. This does mean that a site can be live and fully functional, plus connected to Jetpack, but just that there is an issue with the connection itself.
* This PR focuses predominantly on the Boost columns in the A4A sites dashboard, as that is the main column which didn't behave as expected when there was a Jetpack connection issue
  * In the Boost column, any button or score is now fainter in color in line with other columns for that site, if the site is unreachable. Also, the score or button will now not work.
* I also made sure the `SiteDataField` import was using the imported a8c-for-agencies version as well, as it was still using  the Jetpack Cloud version.
* For future enhancement, we'd likely want to modify the `useFetchTestConnection` hook or create a new hook if we only want to pick up sites that are down.


Note:

* A reverted change included the sites column - the preview pane wouldn't open (intentionally) and some 'Site Unreachable' text would display above the site name. This was reverted as it was only working locally (not in Horizon), ad that was due to the `useFetchTestConnection` hook, which needs a bit more time to investigate. As that functionality was an 'extra' I've just reverted for now and if we want to implement something similar then we can follow up with that.


## Testing Instructions

For the below testing steps, make sure to run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/sites`.

* To test this out, you will need some disconnected or unavailable sites. You may either be able to disconnect a previously connected Jurassic Ninja site, or add a new non-existent site (I wasn't able to get the latter to work). Technically for this to work you may actually only need a site with a broken Jetpack connection.
* You should then see that no columns are clickable for that site on the sites dashboard (except the sites column), and each column visibly appears to be unfunctional, if was previously functional (fainter in color).


Two examples with Boost columns unclickable / lighter in color:
<img width="1174" alt="Screenshot 2024-04-15 at 09 22 11" src="https://github.com/Automattic/wp-calypso/assets/16754605/0382cf94-7082-479e-834f-0f8b400914e2">

<img width="1179" alt="Screenshot 2024-04-15 at 09 22 26" src="https://github.com/Automattic/wp-calypso/assets/16754605/df945239-113b-4af9-8e9c-5ab07265409b">

An example from a site without a connection issue:
<img width="1208" alt="Screenshot 2024-04-15 at 09 23 21" src="https://github.com/Automattic/wp-calypso/assets/16754605/d278a0d7-178b-40b3-aefe-181371e17da0">


## Pre-merge Checklist
